### PR TITLE
Add clang-tidy rosdep keys for debian and ubuntu

### PR DESCRIPTION
**Pull request by jlack1987**
Adding the following keys for `clang-tidy`:

- Ubuntu: https://packages.ubuntu.com/search?keywords=clang-tidy
- Debian Stretch: https://packages.debian.org/stretch/clang-tidy
- Debian Buster: https://packages.debian.org/buster/clang-tidy

There's a gentoo package named `clang`, but i'm very unfamiliar with gentoo and was unable to determine if it has `clang-tidy` contained within, so I left it out. If I need to add it back in let me know. Didn't see anything for Fedora.

### CODE REVIEW

**Review by wjwwood — CHANGES_REQUESTED** [review]
It looks like `clang-tidy` is new in Xenial 16.04, but we require rosdep rules to work on Trusty 14.04 as well.

Can you please use the `'*'` mechanism and specifically provide `trusty: null`?

For example: https://github.com/ros/rosdistro/blob/79242ea09a3703b0f715b5742eea3d7861e103c3/rosdep/base.yaml#L3436-L3438

**Comment by jlack1987** [comment]
Interesting, didn't know that `'*'` functionality existed. Fixed and thanks!

**Review by wjwwood — APPROVED** [review]

**Comment by wjwwood** [comment]
Thanks for iterating. I have to wait for a second review because this is a rosdep rule change, but someone will get to it soon hopefully.

**Pull request merged**
